### DISCUSSION TITLE
Welcome to ctk_theme_builder Discussions!

### DISCUSSION THREAD

**Discussion by avalon60** [avalon60/ctk_theme_builder#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Akascape** [comment]
Hi, @avalon60, you can make a wiki page for the documentation. It will be easier to find and edit them.

**Reply by avalon60** [reply]
> Hi, @avalon60, you can make a wiki page for the documentation. It will be easier to find and edit them.

Hi @Akascape. I'm pretty new to this. I put an entry into the Wiki section and it appeared as a link. This link takes you to the docs. Is there a better way of doing this?

**Reply by Akascape** [reply]
@avalon60 
Yes, there should be a separate link for the docs, you just need to share the Home page link where there will be different content links. And each content can be a new page on the same wiki.

**Comment by Moosems** [comment]
Pretty epic idea to be honest.

**Comment by EngageCosplay** [comment]
Hi, wonder if you can help.
really been struggling with this install.
I believe everything is set but when I try and launch I get this:
Traceback (most recent call last):
  File "C:\Users\n1589h\PycharmProjects\Lab\ctk_theme_builder\ctk_theme_builder.py", line 14, in <module>
    from lib.control_panel import ControlPanel
  File "C:\Users\n1589h\PycharmProjects\Lab\ctk_theme_builder\lib\control_panel.py", line 3, in <module>
    import customtkinter as ctk
ModuleNotFoundError: No module named 'customtkinter'


I know customtkinter is installed as I can run scripts with it fine.

**Reply by Akascape** [reply]
@stevenhunt77 
You will find your answer here:
https://github.com/TomSchimansky/CustomTkinter/issues/1915

**Comment by avalon60** [comment]
Hi @stevenhunt77, did you get this sorted? If not, how are you trying to launch the app?

**Reply by EngageCosplay** [reply]
I did in the end.  Had to create a whole new venv for some reason.  But main thing is I got it running  haha

**Comment by sandipsable506** [comment]
Getting error
TIMEOUT: Waited too long for preview listener!

Ensure that only one instance of CTK Them Builder running on Port 5051, and that no other process using port.

![Capture](https://github.com/avalon60/ctk_theme_builder/assets/146274777/44fbf79b-75d8-4208-95f4-67f975e9b5c9)

Please help

**Reply by avalon60** [reply]
Hello @sandipsable506. You can ignore the "Please upgrade..." that's an erroneous title. 

Please see below:

1. What version of CTK Theme Builder are you using (Tools > About)?
2. What O/S are you running it on?
3. What did you do to get that message to appear?

Thanks.

**Comment by sandipsable506** [comment]
I am using CTk Theme Builder 2.4 version, o/s windows 10, python version 3.11.0
First I was problem on running bat file, when i ran it nothing happen.
So i tried running directly ctk_theme_builder.py file in spyder, Than I was getting ui and when I change them getting multiple same window not getting preview andgetting above error.

**Reply by avalon60** [reply]
I wonder if this is a Windows10 issue. I re-tested Windows11 yesterday without a hitch.

You can't run the ctk_theme_builder.py file without setting the virtual environment, which is what the .bat file does for you.
When you ran the .bat file, was it from a command window? If so were there any messages printed?

I would also be interested to see what the **command _and_ output** looks like, when you ran theme_builder_setup.py.

Thanks,

Clive

**Comment by MawiRecon** [comment]
When I try to run `...\apps\ctk_theme_builder> .\ctk_theme_builder`
VS Code auto opens the .py file, rather than the .bat file launching the theme builder program 

running Python 3.11.8 and VS Code 1.87
Windows 11 
ctk_theme_builder-3.0.1

**Reply by avalon60** [reply]
I suspect you have a Windows file association thing going on. Can you try adding .bat to the end of the command. Ie .\ctk_theme_builder.bat

**Reply by MawiRecon** [reply]
turns out the simplest solution is correct. Yeah that was the case. 
I uninstalled VS and the program opened just fine. 
Redownloaded VS code and unchecked the option to auto open relevant files in VS code. 

Thanks for the response.

**Reply by avalon60** [reply]
Hi @MawiRecon. FYI, I had another person with a similar issue. The file association issue, in respect of VSCode, will be fixed in the upcoming version (possibly released this weekend). 

Version 3.1 will have a few  improvements:

1. Themes can now be Imported / Exported via the *File* menu;
2. The preferences screen has been somewhat redesigned (tidier look);
3. Further Improvements to the installation routines;
4. Clear (runtime) Log function has been added to the Preferences dialogue;
5. A display of the log file size is added to the Preferences dialogue;
6. A new theme "Cobalt" has been added.
